src/components/drawing/index.js

```diff
@@ -69,8 +69,8 @@ drawing.setRect = function(s, x, y, w, h) {
  */
 drawing.translatePoint = function(d, sel, xa, ya) {
     // put xp and yp into d if pixel scaling is already done
-    var x = d.xp || xa.c2p(d.x),
-        y = d.yp || ya.c2p(d.y);
+    var x = d.xp = xa.c2p(d.x);
+    var y = d.yp = ya.c2p(d.y);
```

Comment:
wow, `||` is 4 years old there... but it doesn't look to me like `.xp` and `.yp` are used anywhere except https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/plot.js#L489 which is a near copy of this routine. Take it out? Must have been a perf optimization that we did long ago but is now mostly gone.